### DISCUSSION TITLE
What's the right scope for authorising listEmails()?

### DISCUSSION THREAD

**Discussion by maxi1187** [fattureincloud/api#482] [Q&A] [answered]
Hi, I did research a lot but I still couldn't find what's the right scope for listing the emails?

I have currently enabled `ENTITY_CLIENTS_ALL, PRODUCTS_ALL, ISSUED_DOCUMENTS_INVOICES_ALL, ISSUED_DOCUMENTS_PROFORMAS_ALL, ARCHIVE_ALL, SETTINGS_ALL` in my php project, but I get the NO PERMISSION error.

On https://developers.fattureincloud.it/docs/basics/scopes I see the "emails:r" scope, but I don't see it in the php-sdk, so I'm a bit conufsed.

Also is there a property to know if an invoice has been sent by email in GetIssuedDocumentResponse or how can I figure it out?

Last question which is not clear to me: does the e-invoice get sent to the client after it has been sent to the SDI or do I have to schedule the email?

Many thanks for your help, Max.

**Comment by valmoz** [comment, marked as answer]
Hello @maxi1187 
yes, unfortunately, the **emails:r** scope is missing in our OpenAPI spec and also in our SDKs, we'll add it in the next SDKs releases.
In the meantime, I suggest you use this approach:

- First, try with our [API Reference](https://developers.fattureincloud.it/api-reference) to check if everything is working as expected
- Then, you can add the scope as a custom string in the OAuth 2.0 call. The scopes listed in our SDKs are just a way to make your job easier, but they are just strings you can avoid using if you want. 

At this moment, I confirm there's no such parameter. you can though check if the Webhooks functionality can help you with that, since the **email_sent** events are listed: [Notification Types](https://developers.fattureincloud.it/docs/webhooks/notification-types).

The e-invoice and the email are two different things: 
- The e-invoice is the XML sent to the SDI, it has a dedicated method: [Send the E-Invoice](https://developers.fattureincloud.it/docs/guides/e-invoice-management#mailbox_with_mail-send-the-e-invoice)
- The email is just the PDF of the invoice, this also has a dedicated method: [Send the document](https://developers.fattureincloud.it/docs/guides/invoice-creation#mailbox_with_mail-send-the-document)

The two functionalities are independent, and the two methods must be called separately if needed.

Let us know if you have further questions.

Thanks
Mauro

**Reply by maxi1187** [reply]
Hi, thanks for the quick reply.
Passing a string as a parameter it works great, I didn't think about it, thanks.

Thanks also for clarify the difference between e-invoice and email.

Can I ask one more thing?
I see on the SDK that `listEmails()` has only `company_id` as a parameter.
When I make the call I get only the first 50 emails, but without passing the `per_page` parameter, like for example with `listIssuedDocuments()`, how can I retrieve them all, or how can I sort them? Is that something that will may be implement or am I doing wrong?

**Reply by valmoz** [reply]
Hi @maxi1187 
the listEmails method was not documented properly in our OpenAPI Spec. We need to update our SDKs to describe it correctly, but it will take time. It has both the pagination and the field/fieldset parameters.
You can wait until then, or if you prefer you can implement this method on your own using an HTTP client directly. Unfortunately, you'll need to wait a bit for a new SDK release.

Thanks
Mauro

**Reply by maxi1187** [reply]
Ok no worries thanks for you help
